tests/unit/lms/security_test.py

```diff
@@ -0,0 +1,486 @@
+
+        LTISecurityPolicy.assert_called_once_with()
+        lti_security_policy.forget.assert_called_once_with(mock.sentinel.request)
+        tkt_security_policy.forget.assert_not_called()
+
+    def test_forget_falls_back_on_TktSecurityPolicy(
+        self, policy, lti_security_policy, tkt_security_policy
+    ):
+        lti_security_policy.authenticated_userid.return_value = None
+
+        policy.forget(mock.sentinel.request)
+        tkt_security_policy.forget.assert_called_once_with(mock.sentinel.request)
+        lti_security_policy.forget.assert_not_called()
+
+    def test_identity_delegates_to_LTISecurityPolicy(
+        self,
+        policy,
+        LTISecurityPolicy,
+        lti_security_policy,
+        tkt_security_policy,
+    ):
+        policy.identity(mock.sentinel.request)
+
+        LTISecurityPolicy.assert_called_once_with()
+        lti_security_policy.identity.assert_called_once_with(mock.sentinel.request)
+        tkt_security_policy.identity.assert_not_called()
+
+    def test_permits_delegates_to_LTISecurityPolicy(
+        self,
+        policy,
+        LTISecurityPolicy,
+        lti_security_policy,
+        tkt_security_policy,
+    ):
+        policy.permits(mock.sentinel.request, mock.sentinel.context, "some-permission")
+
+        LTISecurityPolicy.assert_called_once_with()
+        lti_security_policy.permits.assert_called_once_with(
+            mock.sentinel.request, mock.sentinel.context, "some-permission"
+        )
+        tkt_security_policy.permits.assert_not_called()
+
+    @pytest.fixture
+    def policy(self):
+        return SecurityPolicy("TEST_SECRET")
+
+    @pytest.fixture(autouse=True)
+    def TktSecurityPolicy(self, patch):
+        return patch("lms.security.TktSecurityPolicy")
+
+    @pytest.fixture
+    def tkt_security_policy(self, TktSecurityPolicy):
+        return TktSecurityPolicy.return_value
+
+    @pytest.fixture(autouse=True)
+    def LTISecurityPolicy(self, patch):
+        return patch("lms.security.LTISecurityPolicy")
+
+    @pytest.fixture
+    def lti_security_policy(self, LTISecurityPolicy):
+        return LTISecurityPolicy.return_value
+
+
+class TestAuthenticatedUserID:
+    @pytest.mark.parametrize(
+        "lti_user,expected_userid",
+        [
+            (
+                factories.LTIUser(
+                    user_id="sam",
+                    oauth_consumer_key="Hypothesisf301584250a2dece14f021ab8424018a",
+                ),
+                "c2Ft:Hypothesisf301584250a2dece14f021ab8424018a",
+            ),
+            (
+                factories.LTIUser(
+                    user_id="Sam:Smith",
+                    oauth_consumer_key="Hypothesisf301584250a2dece14f021ab8424018a",
+                ),
+                "U2FtOlNtaXRo:Hypothesisf301584250a2dece14f021ab8424018a",
+            ),
+        ],
+    )
+    def test_it(self, lti_user, expected_userid):
+        assert authenticated_userid(lti_user) == expected_userid
```

Comment:
I wonder if this belongs in `TestLTISecurityPolicy` (as a test for `LTISecurityPolicy.authenticated_userid()`) since I think that's the only thing that calls the `authenticated_userid()` function, and I think the function is really a private helper